pallets/did/src/mock.rs

```diff
@@ -17,400 +17,433 @@
+	url_encoded_string.resize(total_length, b'0');
+	Url::Http(
+		HttpUrl::try_from(url_encoded_string.as_ref()).expect("Failed to create default URL with provided length."),
+	)
 }
 
-impl Config for Test {
-	type DidIdentifier = TestDidIdentifier;
-	type Origin = Origin;
-	type Call = Call;
-	#[cfg(feature = "runtime-benchmarks")]
-	type EnsureOrigin = EnsureSigned<TestDidIdentifier>;
-	#[cfg(not(feature = "runtime-benchmarks"))]
-	type EnsureOrigin = did::EnsureDidOrigin<TestDidIdentifier>;
-	type Event = ();
-	type MaxNewKeyAgreementKeys = MaxNewKeyAgreementKeys;
-	type MaxUrlLength = MaxUrlLength;
-	type MaxVerificationKeysToRevoke = MaxVerificationKeysToRevoke;
-	type WeightInfo = ();
+pub fn generate_base_did_creation_details<T: Config>(did: DidIdentifierOf<T>) -> DidCreationDetails<T> {
+	DidCreationDetails {
+		did,
+		new_key_agreement_keys: BoundedBTreeSet::new(),
+		new_attestation_key: None,
+		new_delegation_key: None,
+		new_endpoint_url: None,
+	}
 }
 
-impl ctype::Config for Test {
-	type CtypeCreatorId = TestCtypeOwner;
-	#[cfg(feature = "runtime-benchmarks")]
-	type EnsureOrigin = EnsureSigned<TestDidIdentifier>;
-	#[cfg(not(feature = "runtime-benchmarks"))]
-	type EnsureOrigin = did::EnsureDidOrigin<TestCtypeOwner>;
-	type Event = ();
-	type WeightInfo = ();
+pub fn generate_base_did_update_details<T: Config>() -> DidUpdateDetails<T> {
+	DidUpdateDetails {
+		new_authentication_key: None,
+		new_key_agreement_keys: BoundedBTreeSet::new(),
+		attestation_key_update: DidVerificationKeyUpdateAction::default(),
+		delegation_key_update: DidVerificationKeyUpdateAction::default(),
+		new_endpoint_url: None,
+		public_keys_to_remove: BoundedBTreeSet::new(),
+	}
 }
 
-#[cfg(test)]
-pub(crate) const DEFAULT_ACCOUNT: kilt_primitives::AccountId = kilt_primitives::AccountId::new([0u8; 32]);
-
-const DEFAULT_AUTH_SEED: [u8; 32] = [4u8; 32];
-const ALTERNATIVE_AUTH_SEED: [u8; 32] = [40u8; 32];
-const DEFAULT_ENC_SEED: [u8; 32] = [5u8; 32];
-const ALTERNATIVE_ENC_SEED: [u8; 32] = [50u8; 32];
-const DEFAULT_ATT_SEED: [u8; 32] = [6u8; 32];
-const ALTERNATIVE_ATT_SEED: [u8; 32] = [60u8; 32];
-const DEFAULT_DEL_SEED: [u8; 32] = [7u8; 32];
-const ALTERNATIVE_DEL_SEED: [u8; 32] = [70u8; 32];
-const DEFAULT_URL_SCHEME: [u8; 8] = *b"https://";
-
-pub fn get_did_identifier_from_ed25519_key(public_key: ed25519::Public) -> TestDidIdentifier {
-	MultiSigner::from(public_key).into_account()
+pub fn generate_base_did_details<T: Config>(authentication_key: DidVerificationKey) -> DidDetails<T> {
+	DidDetails::new(authentication_key, BlockNumberOf::<T>::default())
+		.expect("Failed to generate new DidDetails from auth_key due to BoundedBTreeSet bound")
 }
 
-pub fn get_did_identifier_from_sr25519_key(public_key: sr25519::Public) -> TestDidIdentifier {
-	MultiSigner::from(public_key).into_account()
-}
+#[cfg(any(feature = "std", test))]
+pub mod std {
+	#![allow(dead_code)]
+	/// FIXME: Why does clippy not realize the complained functions are used in
```

Comment:
Probably it has to do with the flags used to run Clippy. I guess with `--all-targets` it would not complain anymore.